### PR TITLE
update typescript, linter, use @types instead of typings

### PR DESCRIPTION
**Pull request by clintropolis**
changes:
* updated to latest typescript
* replace typings with @types for dependencies
* remove dts-generator in favor of using @types everywhere
* update tslint, linter config

### CODE REVIEW

File: src/zookeeperBeacon.ts [hunk 1]
```diff
@@ -57,16 +55,18 @@ export class ZookeeperBeacon extends EventEmitter {
...
                         return;
                     }
                     this.client.create(
                         this.config.path + '/' + this.id,
                         new Buffer(JSON.stringify(this.config.payload)),
                         zookeeper.CreateMode.EPHEMERAL,
                         (createErr) => {
-                            if (createErr && createErr.getCode() !== zookeeper.Exception.NODE_EXISTS) {
-                                this.emit(BeaconEvents.ERROR, new Error(`Failed to create node: ${this.config.path}/${this.id} due to: ${err}.`));
+                            if (createErr && ((createErr as any).getCode() !== zookeeper.Exception.NODE_EXISTS)) {
```
Comment (desaianuj) [inline src/zookeeperBeacon.ts:67]: is that casting way preferrable over `<any>createErr`?

File: tslint.json [hunk 2]
```diff
@@ -1,53 +1,58 @@
...
-    "no-arg": true,
-    "no-bitwise": true,
-    "no-console": [true,
-      "debug",
-      "info",
-      "time",
-      "timeEnd",
-      "trace"
+    "ban": [true,
+      ["describe", "only"],
```
Comment (desaianuj) [inline tslint.json:8]: thats awesome

**Comment by desaianuj** [comment]
LGTM except comments 💯

File: src/zookeeperBeacon.ts (hunk 1 above)
Comment (clintropolis) [reply src/zookeeperBeacon.ts:67]: i think so, the linter complains at me

**Pull request merged**